nectar/src/command/trade/event_loop.rs

```diff
@@ -222,99 +222,33 @@ impl EventLoop {
-                            },
-                            secret_hash: exec_swap.hbit.secret_hash,
-                            start_of_swap,
-                            swap_id,
-                            taker: ActivePeer {
-                                peer_id: exec_swap.peer_id,
-                            },
-                        })
-                    }
-                    // Buy
-                    (Role::Alice, setup_swap::SwapProtocol::Herc20Hbit) => {
-                        SwapKind::Herc20Hbit(SwapParams {
-                            hbit_params: crate::swap::hbit::Params::new(
-                                exec_swap.hbit,
-                                bitcoin_transient_sk,
-                            ),
-                            herc20_params: crate::swap::herc20::Params {
-                                asset: exec_swap.herc20.asset.clone(),
-                                redeem_identity: exec_swap.herc20.redeem_identity,
-                                refund_identity: exec_swap.herc20.refund_identity,
-                                expiry: exec_swap.herc20.expiry,
-                                secret_hash: exec_swap.herc20.secret_hash,
-                                chain_id: exec_swap.herc20.chain_id,
-                            },
-                            secret_hash: exec_swap.hbit.secret_hash,
-                            start_of_swap,
-                            swap_id,
-                            taker: ActivePeer {
-                                peer_id: exec_swap.peer_id,
-                            },
-                        })
-                    }
-                    // Sell
-                    (Role::Bob, setup_swap::SwapProtocol::Herc20Hbit) => {
-                        SwapKind::Herc20Hbit(SwapParams {
-                            hbit_params: crate::swap::hbit::Params::new(
-                                exec_swap.hbit,
-                                bitcoin_transient_sk,
-                            ),
-                            herc20_params: crate::swap::herc20::Params {
-                                asset: exec_swap.herc20.asset.clone(),
-                                redeem_identity: exec_swap.herc20.redeem_identity,
-                                refund_identity: exec_swap.herc20.refund_identity,
-                                expiry: exec_swap.herc20.expiry,
-                                secret_hash: exec_swap.herc20.secret_hash,
-                                chain_id: exec_swap.herc20.chain_id,
-                            },
-                            secret_hash: exec_swap.hbit.secret_hash,
-                            start_of_swap,
-                            swap_id,
-                            taker: ActivePeer {
-                                peer_id: exec_swap.peer_id,
-                            },
-                        })
-                    }
+                let swap_kind = match exec_swap.swap_protocol {
```

Comment:
The `SwapParams` struct looks exactly the same so replace with:

```
let swap_params = ...;

let swap_kind = match exec_swap.swap_protocol {
	..::HbitHerc20 => SwapKind::HbitHerc20(swap_params),
	..
};
```

Reply:
Oh, I didn't even realize! Good catch!